xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -247,10 +242,19 @@ func (b *clusterImplBalancer) updateClientConnState(s balancer.ClientConnState)
 		})
 	}
 
+	// Build config for the gracefulswitch balancer. It is safe to ignore JSON
+	// marshaling errors here, since the config was already validated as part of
+	// ParseConfig().
+	cfg := []map[string]any{{newConfig.ChildPolicy.Name: newConfig.ChildPolicy.Config}}
+	cfgJSON, _ := json.Marshal(cfg)
+	parsedCfg, err := gracefulswitch.ParseConfig(cfgJSON)
+	if err != nil {
+		return err
+	}
```

Comment:
Hmm... actually what is the reason for moving this block to be done after handling telemetry labels and drop/request counts?

Reply:
Ah, imo it won't matter when are we building the chilld config for gracefulswitch balancer. Does it?
Anyways, I have reverted that and updated the PR.

Reply:
Generally, it is a good idea to not make a change without a need for it.